### ISSUE TITLE
Does not typecheck input variables for useQuery()

### ISSUE THREAD

**Issue by dko-slapdash** [PabloSzx/graphql-tag-ts#4]
Hi.

Using graphql-tag-ts, we can still pass arbitrary input to variables of e.g. useQuery(), despite the Query being a strongly typed DocumentNode<>. This doesn't protect us from sending garbage to a query.

Example:

```ts
import gql, { DocumentNode } from "graphql-tag-ts";

const Query: DocumentNode<{ some: number }, { arg: string }> = gql`...`;

export function Test() {
  const { data } = useQuery(Query, { variables: { whatever: 42 /* <-- WAT */ } });
  return <div>{data}</div>; // here data is correctly inferred
}
```

P.S.
For useMutation(), both Data and Variables are correctly typechecked, because useMutation() returns a callback and doesn't accept any variables itself as an input. So the problem is in useQuery() only.

P.P.S.
I suspect that I know why this happens: when TS doesn't find a match with useQuery() types mapping in graphql-tag-ts, it falls back to the top level one from @apollo/react-hooks (which effectively accepts arbitrary variables). Looks like there is no way out of it, right?

**Comment by dko-slapdash** [comment]
I think I found a work-around for non-typechecked variables, inspired by https://github.com/microsoft/TypeScript/issues/23689 - here it is:

```ts
  function useQuery<
    TData = any,
    TVariables = OperationVariables
  >(
    query: DocumentNode<TData, TVariables>,
    options?: QueryHookOptions<TData, TVariables>
  ): QueryResult<TData, TVariables>;

  function useQuery<
    TData = any,
    TVariables = OperationVariables
  >(
    query: DocumentNode<TData, TVariables>,
    options: QueryHookOptions<any, any>
  ): ["Expected the following variables type", TVariables];
```

I.e. if the 1st overload doesn't match, and the 2nd overload matches due to `query` being DocumentNode<>, that means we must return some typing error hint to the caller, and it will look like this (VSCode's IntelliSense works fine too):

![image](https://user-images.githubusercontent.com/46383452/85844794-2b51af00-b758-11ea-817a-04c502c7218d.png)

The same technique may be applied in https://github.com/PabloSzx/graphql-tag-ts/issues/3

**Comment by PabloSzx** [comment]
I just released v0.1.16, I hope it's fixes it, since I did some refactoring and updated a couple of types accordingly to @apollo/react-hooks

**Comment by PabloSzx** [comment]
Closing in favor of https://github.com/dotansimha/graphql-typed-document-node and included TypedQueryDocumentNode in graphql-js
